### PR TITLE
fix(broker/bbdo):grpc_stream.proto is bad generated

### PR DESCRIPTION
**Pull request by sechkem**
grpc_stream.proto is bad generated

REFS: MON-147911

### CODE REVIEW

File: broker/grpc/generate_proto.py [hunk 1]
```diff
@@ -177,45 +177,65 @@ class received_protobuf : public io::protobuf<T, Typ> {
...
-                    messages.append([m.group(1), io_protobuf_match.group(1)])
+                    messages.append([m.group(1), io_protobuf_match.group(1),io_protobuf_match.group(2)])
                     io_protobuf_match = None
+                    flag_ignore = True
                 else:
                     io_protobuf_match = re.match(io_protobuf_parser, line)
 
-            if len(messages) > 0:
+                #check if no bbo message have the comment: Ignore 
+                if line.__contains__(ignore_message):
```
Comment (bouda1) [inline broker/grpc/generate_proto.py:206]:
```suggestion
                if ignoere_message in line:
```

File: broker/grpc/generate_proto.py [hunk 2]
```diff
@@ -177,45 +177,65 @@ class received_protobuf : public io::protobuf<T, Typ> {
...
+        line_counter = 0
         with open(join(directory, file)) as proto_file:
             messages = []
             io_protobuf_match = None
             for line in proto_file.readlines():
+                line_counter += 1
                 m = re.match(message_parser, line)
                 if m is not None and io_protobuf_match is not None:
-                    messages.append([m.group(1), io_protobuf_match.group(1)])
+                    messages.append([m.group(1), io_protobuf_match.group(1),io_protobuf_match.group(2)])
```
Comment (bouda1) [inline broker/grpc/generate_proto.py:199]:
```suggestion
                    messages.append([m.group(1), io_protobuf_match.group(1), io_protobuf_match.group(2)])
```

File: broker/grpc/generate_proto.py [hunk 3]
```diff
@@ -177,45 +177,65 @@ class received_protobuf : public io::protobuf<T, Typ> {
...
                     io_protobuf_match = None
+                    flag_ignore = True
                 else:
                     io_protobuf_match = re.match(io_protobuf_parser, line)
 
-            if len(messages) > 0:
+                #check if no bbo message have the comment: Ignore 
+                if line.__contains__(ignore_message):
+                    flag_ignore = True
+                if flag_ignore == True and m is not None:
```
Comment (bouda1) [inline broker/grpc/generate_proto.py:208]:
```suggestion
                if flag_ignore and m is not None:
```

File: broker/grpc/generate_proto.py [hunk 4]
```diff
@@ -177,45 +177,65 @@ class received_protobuf : public io::protobuf<T, Typ> {
...
+                    flag_ignore = False
+                elif flag_ignore == False and m is not None :
```
Comment (bouda1) [inline broker/grpc/generate_proto.py:210]:
```suggestion
                elif not flag_ignore and m is not None : 
```

File: broker/grpc/generate_proto.py [hunk 5]
```diff
@@ -177,45 +177,65 @@ class received_protobuf : public io::protobuf<T, Typ> {
...
-                stream_content, &grpc_event_type::{lower_mess}_,
-                &grpc_event_type::mutable_{lower_mess}_);
+                message_save += messages
+#sort the message with index (io_protobuf_match.group(2))
+message_save.sort(key=lambda x: int(x[2]))
+for mess, id, index in message_save:
+    # proto file
+    file_message_centreon_event += f"        {mess} {mess}_ = {index};\n"
+    # count index : needed for opentelemetry
+    one_of_index +=1
```
Comment (bouda1) [inline broker/grpc/generate_proto.py:224]:
```suggestion
    one_of_index += 1
```

File: broker/grpc/generate_proto.py (hunk 1 above)
Comment (sechkem) [reply broker/grpc/generate_proto.py:206]: will fix

File: broker/grpc/generate_proto.py [hunk 6]
```diff
@@ -177,45 +177,66 @@ class received_protobuf : public io::protobuf<T, Typ> {
...
             messages = []
             io_protobuf_match = None
             for line in proto_file.readlines():
+                line_counter += 1
                 m = re.match(message_parser, line)
                 if m is not None and io_protobuf_match is not None:
-                    messages.append([m.group(1), io_protobuf_match.group(1)])
+                    messages.append([m.group(1), io_protobuf_match.group(1), io_protobuf_match.group(2)])
                     io_protobuf_match = None
+                    flag_ignore = True
```
Comment (jean-christophe81) [inline broker/grpc/generate_proto.py:201]: flag_ignore = False?

File: broker/grpc/CMakeLists.txt [hunk 7]
```diff
@@ -89,6 +89,8 @@ add_custom_command(
     grpc_stream.proto
   VERBATIM)
 
+  add_custom_target("grpc_stream.proto" DEPENDS ${MODULE_DIR}/grpc_stream.proto)
+
```
Comment (bouda1) [inline broker/grpc/CMakeLists.txt:93]: seems not needed

Comment (bouda1) [inline broker/grpc/CMakeLists.txt:93]: if not needed, you can remove that.

File: broker/grpc/generate_proto.py (hunk 6 above)
Comment (sechkem) [reply broker/grpc/generate_proto.py:201]: in 201 : Already found an io_protobuf, similar to found the comment /* Ignore */ .

**Pull request merged**